misc.go

```diff
@@ -886,6 +866,23 @@ var hintTokenMap = map[string]int{
 	"MATERIALIZATION": hintMaterialization,
 }
 
+type featureMap map[string]struct{}
+
+func (m featureMap) containsAll(features []string) bool {
+	for _, f := range features {
+		if _, found := m[f]; !found {
+			return false
+		}
+	}
+	return true
+}
+
+// featureMap is a set of feature IDs supported by current parser.
+var featuresMap = featureMap{
```

Comment:
Let the user to fill this map. You can provide a function like `RegistFeature`.
Or just let the parser return the parsed feature list strings and let the user to handle them.